agent/engine/docker_task_engine.go

```diff
@@ -945,6 +959,24 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	return metadata
 }
 
+// Log driver options type, tag and fluentd-address are added
+// For reference - https://docs.docker.com/config/containers/logging/fluentd/
+func getLogRouterConfig(task *apitask.Task, container *apicontainer.Container, hostConfig *container.HostConfig, engine *DockerTaskEngine) container.LogConfig {
+	seelog.Infof("Since the value for logrouter in container is %s, we modify LogConfig values in hostconfig to support log router driver", logConfigType)
+	fields := strings.Split(task.Arn, "/")
+	taskID := fields[len(fields)-1]
+	tag := container.Name + "-" + taskID
+	fluentd := engine.cfg.DataDirOnHost + dataLogDriverPath + taskID + dataLogDriverSocketPath
+	logConfig := hostConfig.LogConfig
```

Comment:
I think what you want to do here is something like:

logConfig := LogConfig{} // new instance of log config object

But correct me if I'm mistaken :)